src/codeu/chat/server/Server.java

```diff
@@ -207,6 +207,24 @@ private boolean onMessage(InputStream in, OutputStream out) throws IOException {
       Serializers.collection(Message.SERIALIZER).write(out, messages);
       
 
+    } else if (type == NetworkCode.GET_MESSAGES_BY_RANGE_REQUEST) {
```

Comment:
This whole large if/else block could be replaced with a switch statement

switch (type) {
 case GET_MESSAGES_BY_RANGE_REQUEST: ...
 case GET_MESSAGES_BY_TIME_REQUEST: ...
default: <error>
}